src/Headers/ContentSecurityPolicy/Header.php

```diff
@@ -0,0 +1,83 @@
+<?php declare(strict_types = 1);
+
+namespace Pd\SecurityHeaders\Headers\ContentSecurityPolicy;
+
+final class Header implements \Pd\SecurityHeaders\Headers\IHeader
+{
+
+	/**
+	 * @var Directive|null
+	 */
+	private $defaultSrc;
+
+	/**
+	 * @var array|Directive[]
+	 */
+	private $directives = [];
+
+	/**
+	 * @var bool
+	 */
+	private $blockAllMixedContent;
+
+
+	public function getName(): string
+	{
+		return 'Content-Security-Policy';
+	}
+
+
+	public function getValue(): string
+	{
+		$directives[] = $this->defaultSrc ? ('default-src ' . $this->defaultSrc) : '';
+
+		foreach ($this->directives as $directiveName => $directiveValues) {
+			$directives[] = $directiveName . ' ' . $directiveValues;
+		}
+
+		return \implode('; ', $directives) . ($this->blockAllMixedContent ? '; block-all-mixed-content' : '');
```

Comment:
Je potřeba mít `block-all-mixed-content` zvlášť? Nemůže se proto udělat direktiva jako pro ostatní?

Reply:
Ono to může být jen jednou, navíc to není direktiva obsahující další hodnoty, je to string bez dalších hodnot. Leda bych to udělal podobně jako direktivu, která ale nemá další metody. Musel bych potom hlídat, že tam není přidaná vícekrát, takže `$header->setBlockAllMixedContent(new BlockAllMixedContent);`. To mi už nedává smysl, protože to je bez hodnot.

Reply:
Takže tato direktiva tam nemůže být vícekrát, ale ostatní mohou?
Vím, že tato direktiva nemá žádné další "argumenty".  Potom jsou ale i další takové direktivy `upgrade-insecure-requests` nebo `disown-opener`, které jsou mimochodem také užitečné.
Určitě se to dá řešit tak, jak jsi to udělal ty, ale jenom mě napadlo, že by to šlo udělat třeba nějak lépe.

Reply:
Všechny tam mohou být jen jednou, to je pravda. To teď nekontroluji.